PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,433 @@
+    lta_convert --inlta $T1w_dir_long/xfms/T1w_cross_to_norm.lta --outlta $T1w_dir_long/xfms/norm_to_T1w_cross.lta --invert
+    #4. Combine TP->template and orig->TP transforms to get TP->TP(HCP template) transform.
+    mri_concatenate_lta $T1w_dir_long/xfms/T1w_cross_to_template.lta $T1w_dir_long/xfms/norm_to_T1w_cross.lta $T1w_dir_long/xfms/T1w_cross_to_T1w_long.lta	
+    #5. convert the final TP_T1w->TP_T1w(HCP template) LTA transform to .mat(FSL).
+    lta_convert --inlta $T1w_dir_long/xfms/T1w_cross_to_T1w_long.lta --src $T1w_cross --trg $T1w_cross --outfsl $T1w_dir_long/xfms/T1w_cross_to_T1w_long.mat
+else #Template mode
+    :
+fi    
+#end block
+
+#################################################################################################################
+# The next block resamples T1w from native to the longitudinal acpc_dc template space, using cross-sectional 
+# readout distortion correction warp.
+
+if (( TemplateProcessing == 0 )); then 
+    log_Msg "Timepoint $Timepoint_long: combining warps and resampling T1w_acpc_dc to longitudinal template space"
+    T1w_cross_gdc_warp=$T1w_dir_cross/xfms/${T1wImage}1_gdc_warp.nii.gz
+    if [ ! -f "$T1w_cross_gdc_warp" ]; then
+        log_Msg "NOT USING GRADIENT DISTORTION CORRECTION WARP"
+        create_zero_warp $T1w_dir_cross/${T1wImage}1_gdc.nii.gz $T1w_cross_gdc_warp
+    fi
+
+    #first, create the warp transform.
+    convertwarp --premat="$T1w_dir_cross/xfms/acpc.mat" --ref=$MNI_08mm_template \
+        --warp1=$T1w_dir_cross/xfms/T1w_dc.nii.gz --postmat=$T1w_dir_long/xfms/T1w_cross_to_T1w_long.mat --out=$T1w_dir_long/xfms/T1w_acpc_dc_long.nii.gz
+    #second, apply warp.
+    applywarp --interp=spline -i $T1w_dir_cross/T1w.nii.gz -r $MNI_08mm_template \
+        -w $T1w_dir_long/xfms/T1w_acpc_dc_long.nii.gz -o $T1w_dir_long/T1w_acpc_dc.nii.gz
+else 
+    :
+fi
+#end block
+
+
+##############################################################################################################
+# The next block creates the warp from T2w to template space, including readout distortion correction, and applies it
+
+T2w_dir_cross=$StudyFolder/$Timepoint_cross/T2w
+T2w_dir_long=$StudyFolder/$Timepoint_long/T2w
+T2w_dir_template=$StudyFolder/$Subject.long.$Template/T1w
+Use_T2w=1
+
+if (( TemplateProcessing == 0 )); then 
+
+
+    mkdir -p $T2w_dir_long/xfms
+
+    #This uses combined transform from T1w to T2w, whish is always generated. Should replace the if statements below.
+    if [ -f "$T1w_dir_cross/xfms/T2w_reg_dc.nii.gz" ]; then
+        log_Msg "Timepoint $Timepoint_long: warping T2-weighted image to T2w_acpc_dc in longitudinal template space"
+        convertwarp --premat=$T2w_dir_cross/xfms/acpc.mat --ref=$MNI_08mm_template \
+            --warp1=$T1w_dir_cross/xfms/T2w_reg_dc.nii.gz --postmat=$T1w_dir_long/xfms/T1w_cross_to_T1w_long.mat --out=$T2w_dir_long/xfms/T2w2template.nii.gz
```

Comment:
This seems to be missing the final FreeSurfer BBR refinement of the T2w registration: https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/PostFreeSurfer/scripts/CreateMyelinMaps.sh#L139-L140
https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/PostFreeSurfer/PostFreeSurferPipeline.sh#L156

Reply:
If you use the full transform above, that should be more accurate.

Reply:
will look into this.